common/reviews/api/heft.api.md

```diff
@@ -248,8 +249,9 @@ export interface IHeftActionConfigurationOptions {
 }
 
 // @beta
-export interface IHeftBaseParameter {
+export interface IHeftBaseParameter<TValue> {
     readonly actionAssociated: boolean;
```

Comment:
Suggested doc clarification:

OLD:
```ts
    /**
     * The currently selected action was associated with the parameter.
     */
    readonly actionAssociated: boolean;
    /**
     * The parameter was specified on the command line.
     */
    readonly valueProvided: boolean;
```
NEW:
```ts
    /**
     * If true, then the user has invoked Heft with a command-line action that supports this parameter
     * (as defined by the {@link IWhateverItIsNow.associatedActionNames} option).
     * 
     * @remarks
     * For example, if  `build`  is one of the associated action names for `--my-integer-parameter`,
     * then `actionAssociated` will be true if the user invokes `heft build`.
     * 
     * To test whether the parameter was actually included (e.g. `heft build --my-integer-parameter 123`),
     * use the {@link IHeftBaseParameter.valueProvided} property.
     */
    readonly actionAssociated: boolean;

    /**
     * If true, then the user included this parameter on the command-line when invoking Heft.
     * The value can be retrieved from {@link IHeftBaseParameter.value}.
     * 
     * @remarks
     * If `valueProvided` is true, then `actionAssociated` must have been true, otherwise
     * the parameter is not allowed to be used.
     */
    readonly valueProvided: boolean;
```

Reply:
I applied most of this for actionAssociated.